packages/tasit-action/src/contract/Contract.test.js

```diff
@@ -288,94 +285,86 @@ describe("TasitAction.Contract", () => {
           const value = await sampleContract.getValue();
           expect(value).to.equal(rand);
         }
-        confirmationFakeFn();
-      };
+      });
+
+      const errorListener = sinon.fake();
 
       action.on("error", errorListener);
       action.on("confirmation", confirmationListener);
 
-      await mineBlocks(provider, 6);
+      await mineBlocks(provider, 7);
 
-      expect(confirmationFakeFn.callCount).to.equal(6);
-      expect(errorFakeFn.called).to.be.false;
+      expect(confirmationListener.callCount).to.equal(6);
+      expect(errorListener.called).to.be.false;
     });
 
     it("should change contract state and trigger confirmation event - late subscription", async () => {
       action = sampleContract.setValue(rand);
       await action.send();
-
       await action.waitForOneConfirmation();
 
       await mineBlocks(provider, 5);
 
-      const confirmationFakeFn = sinon.fake();
-      const errorFakeFn = sinon.fake();
-
-      const errorListener = message => {
-        errorFakeFn();
-      };
+      const errorListener = sinon.fake();
 
-      const confirmationListener = async message => {
+      const confirmationListener = sinon.fake(async message => {
         const { data } = message;
         const { confirmations } = data;
 
-        if (confirmations == 7) {
+        if (confirmations >= 7) {
```

Comment:
With the way our test is written, it never will get anywhere close to 7 confirmations, right?

Reply:
Not sure if I've got your question. The code should work that way:
```javascript
action = sampleContract.setValue(rand);
await action.send();
await action.waitForOneConfirmation();

// 2nd to 6th confirmations
await mineBlocks(provider, 5);

const errorListener = sinon.fake();

const confirmationListener = sinon.fake(async message => {
  const { data } = message;
  const { confirmations } = data;

  if (confirmations >= 7) {
    action.off("confirmation");

    const value = await sampleContract.getValue();
    expect(value).to.equal(rand);
  }
});

// Subscription to the events
action.on("error", errorListener);
action.on("confirmation", confirmationListener);

// Not sure why but some times one confirmation (6 on that case) isn't caught 
// Because of that, we're mining 2 blocks and guaranteeing at least 7 confirmations
await mineBlocks(provider, 2);

// Non-deterministic
// Could be changed to expect(confirmationListener.callCount).to.be.within(1,2);
// Not sure why it's happen but I think that could be related to the ganache tx 1st conformation.
expect(confirmationListener.callCount).to.be.at.least(1);
expect(errorListener.called).to.be.false;
```

Reply:
And I've changed `==` to `>=` because I've seen before the polling jump confirmations before. Combined with the unsubscription, that listener is working as something like `action.once('at-least-7-confs')`.

Reply:
Oh right, I forgot about the blocks we were mining for the 2nd to 6th confirmations farther up.

Never mind. Thanks for explaining!

Reply:
Feel free to resolve this

Reply:
Resolving this.